### PR TITLE
Option to not connect to ext services

### PR DESCRIPTION
**Pull request by MrCreosote**
also close arango client properly

### CODE REVIEW

**Comment by MrCreosote** [comment]
The codecov UI has really become unreliable, it's a different problem almost every time I go there
image

File: src/service/app.py [hunk 1]
```diff
@@ -60,37 +60,52 @@ def create_app(noop=False):
...
+                await sysdb.create_database(cfg.arango_db)
+            except aioarango.exceptions.DatabaseCreateError as e:
+                if e.error_code != ARANGO_ERR_NAME_EXISTS:  # ignore, db exists
+                    raise
+        db = await _get_arango_db(cli, cfg.arango_db, cfg)
+        storage = await create_storage(db, create_collections_on_startup=cfg.create_db_on_startup)
+        return cli, storage
+    except:
+        await cli.close()
+        raise
```
Comment (Tianhao-Gu) [inline src/service/app.py:103]: will it provide us with enough traceback to troubleshoot?

**Review by Tianhao-Gu — APPROVED** [review]
Just have a few questions. But overall looks good to me.

File: src/service/app.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 from src.service.arg_checkers import contains_control_characters
 from src.service.config import CollectionsServiceConfig
 from src.service.http_bearer import KBaseHTTPBearer, KBaseUser
-from src.service.storage_arango import create_storage
+from src.service.storage_arango import create_storage, ARANGO_ERR_NAME_EXISTS, ArangoStorage
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/service/app.py:26]: The reason for `create_storage` is that async operations need to happen at the initialization of `ArangoStorage`, but __init__ can't be aysnc, so it has to be a different function. I happened to make it a module function, but it just as easily could be a `@classmethod` so the signature would be something like `ArangoStorage.create()`. To me it doesn't make a big difference, but if you think it's important I have no issues swapping to a class method

File: src/service/app.py (hunk 1 above)
Comment (MrCreosote) [reply src/service/app.py:103]: I don't understand the concern - the `raise` will raise the exact same error the was caught so we're not losing any info...?

**Comment by MrCreosote** [comment]
Since you approved I'm going to go ahead and merge, but we can continue with the conversations and I'll make any changes coming out of that in a later PR

File: src/service/app.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/service/app.py:103]: yea. yea. you are right.

**Comment by MrCreosote** [comment]
Oop, you beat me to it :P

File: src/service/app.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/service/app.py:26]: I actually think the classmethod is a bit more readable on second thought, I'll probably do that at some point

**Pull request merged**